crypto/merkle/proof_op.go

```diff
@@ -68,7 +68,10 @@ func (poz ProofOperators) Verify(root []byte, keypath string, args [][]byte) (er
 	return nil
 }
 
-func (poz ProofOperators) VerifyKeys(root []byte, keys [][]byte, args [][]byte) (err error) {
+// VerifyFromKeys performs the same verification logic as the normal Verify
+// method, except it does not perform any processing on the keypath. This is
+// useful when using keys that have split or escape points as a part of the key.
+func (poz ProofOperators) VerifyFromKeys(root []byte, keys [][]byte, args [][]byte) (err error) {
```

Comment:
while I did add some docs to the new methods, I didn't add docs to the existing methods from which they were copy paste forked. Imo, we should upstream those instead of adding them because 1) doesn't increase our diff to tendermint and 2) good habit

For similar reasons, I didn't ad godocs, but definitely still can. If we want those, I think we should first upstream docs for the current methods, then add ours so they make more sense.

Reply:
I think just adding documentation to the new methods is sufficient. People are more likely to examine the diff between this and the upstream and thus want to know why certain things were added. 

I agree with trying to upstream these changes and let the tendermint team decide if they want to add godocs to the existing methods.

I actually think we should have the discussion as to whether this belongs in Tendermint at all or rather in the SDK